### ISSUE TITLE
Add trigger that won't trigger until all pods have completed their deployment and are "ready"

### ISSUE THREAD

**Issue by scott-david-walker** [kubeshop/testkube#2905]
**Describe the enhancement you'd like to see**

I'd like to be able to automatically run the tests but only after a deployment has completed OR all pods within a deployment have completed.

Currently, I deploy via helm with ArgoCD. I have a set trigger that listens to the deployment and runs a test when the image is updated. However, because the pods could take x amount of time to deploy, I want to wait until that has completed before running the tests (I don't want the tests to run against old pods). I have a workaround for this which is add a delay to a test suite starting but that only solves a part of the issue.

Second issue is that if I deploy a trigger and a test together then the test won't run the first time. This is probably due to a race condition where the trigger/test isn't in place yet.

**Additional context**

We have a pull request that creates an environment so triggers/tests not being available at a set time is quite likely

**Comment by vsukhin** [comment]
thank you @scott-david-walker we will discuss it with the team

**Comment by TheBrunoLopes** [comment]
Great insights as usual @scott-david-walker !
I would love to pick your brain to plan out our next features and who's better than a user for that ? xD

Would you spare a few minutes to go on a call https://calendly.com/bruno-at-kubeshop/30min-1 ?

**Comment by scott-david-walker** [comment]
Hi @TheBrunoLopes 
Would be nice to have that discussion, I’ve arranged something for next week. 

Cheers 😊

**Comment by TheBrunoLopes** [comment]
Thank you! looking forward to it

**Comment by vsukhin** [comment]
For first issue we can support this conditions https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#complete-deployment
For second issue we need to make some tests

**Comment by vsukhin** [comment]
@scott-david-walker 
For the first issued we introduce test trigger conditions, please find details there https://kubeshop.github.io/testkube/using-testkube/triggers
For second issue, please make you sure you apply deployment after test and testtrigger creation. Last 2 can be created in any order

**Comment by TheBrunoLopes** [comment]
Gj Vlad!
@scott-david-walker let us know how it goes and if we can assist in any way.

**Comment by scott-david-walker** [comment]
Hi @TheBrunoLopes 

I've just tested this and it works perfectly but I've found 1 small issue that I'm hoping you can advise on.

If you have a minimum number of replicas and deploy a new image, a deployment never actually goes out of the ready state as it only terminates a pod when another one has already been created. Meaning both `Available` and `Progressing` types are always classed as true. The only way I can see around this would be to actively check the "reason" field as a condition? But I don't think that was implemented as a part of this pr?  Unless there's another way that I'm missing?

Thanks
Scott

**Comment by vsukhin** [comment]
hey @scott-david-walker we can check a reason as well. should be relatively small fix. you can create an issue
